packages/interbit-covenant-tools/src/rootCovenant/covenants/manifestCovenant/applyJoinChanges.js

```diff
@@ -0,0 +1,170 @@
+const _ = require('lodash')
+const {
+  startConsumeState,
+  startProvideState,
+  authorizeReceiveActions,
+  authorizeSendActions
+} = require('interbit-covenant-utils')
+const {
+  redispatch,
+  actionTypes: coreActionTypes
+} = require('../../../coreCovenant')
+const {
+  stopConsumeState,
+  stopProvideState,
+  revokeSendActions,
+  revokeReceiveActions
+} = require('../../../coreCovenant/helpers/revokeJoin')
+const { JOIN_TYPES } = require('../../../constants')
```

Comment:
These should be exposed to user covenants as coreCovenant.actionCreators, so they appear as peers of the equivalent start/authorize actions.

Reply:
These are not actually action types. They are constants from config state path . eg. `sendActionsTo`, `receiveActionsFrom` I think we would be better served with a selector.

Reply:
I would have recognized them as constants if they had been in uppercase.